python/caffe/_caffe.cpp

```diff
@@ -275,7 +275,7 @@ BOOST_PYTHON_MODULE(_caffe) {
   bp::class_<vector<bool> >("BoolVec")
     .def(bp::vector_indexing_suite<vector<bool> >());
 
-  import_array();
+  import_array1();
```

Comment:
here is what goes wrong:
```/usr/include/python3.4m/numpy/__multiarray_api.h:1708:35: error: return-statement with a value, in function returning 'void' [-fpermissive]
 #define NUMPY_IMPORT_ARRAY_RETVAL NULL```
The boost python function ```BOOST_PYTHON_MODULE(_caffe)``` is always defined as ```void BOOST_PP_CAT(init_module_, name)()``` which translates to something like ```void init_module__caffe()```. This is irrespective of the python version (which is good).
Numpy switches out the return value of ```import_array``` from void to NULL, when switching from python2 to python3, for whatever reason. ```import_array1``` fixes this.